### PR TITLE
see #548, clean document from geo field on document deletion

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

File: src/module.c [hunk 1]
```diff
@@ -500,7 +500,19 @@ int DeleteCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   if (argc == 4 && RMUtil_StringEqualsCaseC(argv[3], "DD")) {
     delDoc = 1;
   }
-  int rc = DocTable_Delete(&sp->docs, MakeDocKeyR(argv[2]));
+  RSDocumentKey docId = MakeDocKeyR(argv[2]);
+  RedisSearchCtx sctx = SEARCH_CTX_STATIC(ctx, sp);
+  t_docId id = DocTable_GetId(&sp->docs, docId);
+  for (int i = 0; i < sp->numFields; ++i) {
```
Comment (mnunberg) [inline src/module.c]: size_t

File: src/module.c [hunk 2]
```diff
@@ -500,7 +500,19 @@ int DeleteCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
...
+    FieldSpec *fs = sp->fields + i;
+    if (fs->type != FIELD_GEO) {
+      continue;
+    }
+    GeoIndex gi = {.ctx = &sctx, .sp = fs};
+    GeoIndex_Remove(&gi, id);
+  }
+
+  int rc = DocTable_Delete(&sp->docs, docId);
   if (rc == 1) {
```
Comment (mnunberg) [inline src/module.c]: if (rc) {

Comment (MeirShpilraien) [reply src/module.c]: @mnunberg you know I did not touched this line in this PR, but I will change it anyway ..

Comment (mnunberg) [reply src/module.c]: Sorry, I missed that :)

**Pull request merged**